### ISSUE TITLE
Array of array issue

### ISSUE THREAD

**Issue by mnguyen972** [midknight41/map-factory#56]
```js
var createMapper = require("map-factory")

var source = {
    foo: [
        {"name": "a", "things": ["a1", "a2"]},
        {"name": "b", "things": ["b1", "b2"]}
    ]
}
const mapper = createMapper();

mapper
  .map("foo[].name").to("bar[].label")
  .map("foo[].things[]").to("bar[].values[]");
 
const result = mapper.execute(source);
```

...which gives us this:
```js
bar: [{
        label: "a",
        values: [
            ["a1", "a2", "b1", "b2"]
        ]
    },
    {
        label: "b"
    }
]
```
...but what's required is this:
```js
bar: [{
        label: "a",
        values: ["a1", "a2"]
    },
    {
        label: "b",
        values: ["b1", "b2"]
    }
]
```

**Comment by midknight41** [comment]
I'm on holiday at the moment. Will have a look when I return.

**Comment by mnguyen972** [comment]
The latest object-mapper library fixes this issue.  I overrided the object-mapper folder and seems to be working now.

**Comment by midknight41** [comment]
Thanks. Looking into it now.

**Comment by midknight41** [comment]
Does object-mapper fix this? I tried this with it and got this result:

```js
const mapper = require("object-mapper");

const src = {
  foo: [
    { "name": "a", "things": ["a1", "a2"] },
    { "name": "b", "things": ["b1", "b2"] }
  ]
}

const map = {
  "foo[].name": "bar[].label",
  "foo[].things[]": "bar[].values[]"
}

const result = mapper(src, map);
console.log(result);

/*
{
  "bar": [
    { "label": "a", "values": [[["a1", "a2"], ["b1", "b2"]]] },
    { "label": "b" }
  ]
}
*/
```
Granted this behaviour isn't what you were after but, unless I'm mistaken, it is consistent with object-mapper.

I am working on an ```unwind()``` feature that will flatten the structure to get what you want as it turns out...

**Comment by midknight41** [comment]
Published to npm as 3.0.0.
